### PR TITLE
Delayed Visualization Attaching

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR:
* adds notion of readiness to execution context and executed graph
* delays attaching visualization until graph is ready;
* refactors portion on visualization handling logic into a separate module.

### Checklist
Please include the following checklist in your PR:

- [ ] The `CHANGELOG.md` was updated with the changes introduced in this PR.
- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [ ] All code has automatic tests where possible.
- [ ] All code has been profiled where possible.
- [ ] All code has been manually tested in the IDE.
- [ ] All code has been manually tested in the "debug/interface" scene.
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://airtable.com/shr7KPRypRpanF7TO).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://airtable.com/shr7KPRypRpanF7TO).

### CODE REVIEW

File: src/rust/ide/src/ide/integration/visualization.rs [hunk 1]
```diff
@@ -0,0 +1,667 @@
...
+use crate::sync::Synchronized;
+use crate::controller::ExecutedGraph;
+
+use futures::channel::mpsc::UnboundedReceiver;
+use futures::future::ready;
+use ide_view::graph_editor::component::visualization;
+use ide_view::graph_editor::SharedHashMap;
+use ide_view::graph_editor::component::visualization::instance::ContextModule;
+use ide_view::graph_editor::component::visualization::Metadata;
+
```
Comment (farmaazon) [inline src/rust/ide/src/ide/integration/visualization.rs:18]: Missing header.

File: src/rust/ide/src/ide/integration/visualization.rs [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use utils::test::traits::*;
+    use crate::model::execution_context::ModuleQualifiedName;
```
Comment (farmaazon) [inline src/rust/ide/src/ide/integration/visualization.rs]:
```suggestion
    use super::*;
    use utils::test::traits::*;
    
    use crate::model::execution_context::ModuleQualifiedName;
```

File: src/rust/ide/src/model/execution_context/synchronized.rs [hunk 3]
```diff
@@ -89,14 +112,27 @@ impl ExecutionContext {
...
+        match notification {
+            Notification::Completed => {
+                if !self.model.is_ready.replace(true) {
+                    WARNING!("Context {self.id} Became ready");
+                }
+            }
+            Notification::ExpressionUpdates(updates) => {
+                self.model.computed_value_info_registry.apply_updates(updates);
+            }
+        }
```
Comment (farmaazon) [inline src/rust/ide/src/model/execution_context/synchronized.rs:126]: If it is the ExecutionContext which both defines and handles the notification, why we cannot just create functions "handle_expression_updates" and "mark_as_ready"?

File: src/rust/ide/src/model/project.rs [hunk 4]
```diff
@@ -124,6 +124,10 @@ pub trait API:Debug {
     fn urm(&self) -> Rc<model::undo_redo::Manager>;
 }
 
+// Note: Needless lifetimes
+// ~~~~~~~~~~~~~~~~~~~~~~~~
+// See Note: [Needless lifetimes] is `model/project.rs`.
+
```
Comment (farmaazon) [inline src/rust/ide/src/model/project.rs]: Why there is such note? I don't see any "allow" needing explanation.

File: src/rust/ide/src/model/project.rs [hunk 5]
```diff
@@ -239,10 +243,15 @@ pub mod test {
...
     pub fn expect_qualified_name
     (project:&mut MockAPI, name:&QualifiedName) {
         let name = name.clone();
         project.expect_qualified_name().returning_st(move || name.clone());
     }
+
+    pub fn expect_qualified_module_name
+    (project:&mut MockAPI) {
+        let name = project.qualified_name();
+        project.expect_qualified_module_name().returning_st(move |path:&model::module::Path| path.qualified_module_name(name.clone()));
```
Comment (farmaazon) [inline src/rust/ide/src/model/project.rs]: long line.

File: src/rust/ide/src/model/project/synchronized.rs [hunk 6]
```diff
@@ -15,7 +16,7 @@ use crate::transport::web::WebSocket;
 use enso_protocol::binary;
 use enso_protocol::binary::message::VisualisationContext;
 use enso_protocol::language_server;
-use enso_protocol::language_server::{CapabilityRegistration, ContentRoot};
+use enso_protocol::language_server::{CapabilityRegistration, ContentRoot, ExpressionUpdates};
```
Comment (farmaazon) [inline src/rust/ide/src/model/project/synchronized.rs]: imports.

**Review by farmaazon — APPROVED** [review]

File: src/rust/ide/src/model/project.rs (hunk 4 above)
Comment (mwu-tow) [reply src/rust/ide/src/model/project.rs]: This should have been placed in the execution context.

File: CHANGELOG.md [hunk 7]
```diff
@@ -9,8 +9,12 @@
 - [Visualization previews are disabled.][1817] Previously, hovering over a
   node's output port for more than four seconds would temporarily reveal the
   node's visualization. This behavior is disabled now.
+- [Visualizations will be attached after project is ready.][1825] This addresses
```
Comment (MichaelMauderer) [inline CHANGELOG.md]: This needs to move up to the `Next Release` section.

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**